File: Java/Step04/BOJ_1546.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class BOJ_1546 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        List <Integer> score = new ArrayList<Integer>();
        for (int i=0; i<N; i++){
            int A = sc.nextInt();
            score.add(A);
        }
        int M = Collections.max(score);
        double num = 0;
        double total = 0;
        for(double i : score){
            num = i/M*100;
            total += num;
        }
        System.out.println(total/N);
        sc.close();
    }
    
}

File: C/Step04/BOJ_1546.c
#include <stdio.h>

int main(void) {
    int n;
    scanf("%d", &n);

    int score[1000];
    int max = 0;
    double sum = 0;

    for (int i = 0; i < n; i++) {
        scanf("%d", &score[i]);
        if (score[i] > max) {
            max = score[i];
        }
    }

    for (int i = 0; i < n; i++) {
        sum += (double)score[i] / max * 100;
    }

    printf("%.6f\n", sum / n);

    return 0;
}